source/Halibut/Transport/Streams/NetworkTimeoutStream.cs

```diff
@@ -161,30 +150,105 @@ public override void EndWrite(IAsyncResult asyncResult)
             }
         }
 
-        public override void Flush() => inner.Flush();
+        public override void Flush()
+        {
+            ThrowIfAlreadyTimedOut();
+            
+            inner.Flush();
+        }
 
-        public override int Read(byte[] buffer, int offset, int count) => inner.Read(buffer, offset, count);
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            return TryCloseOnTimeout(() => inner.Read(buffer, offset, count));
+        }
 
-        public override long Seek(long offset, SeekOrigin origin) => inner.Seek(offset, origin);
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            ThrowIfAlreadyTimedOut();
+            
+            return inner.Seek(offset, origin);
+        }
 
-        public override void SetLength(long value) => inner.SetLength(value);
+        public override void SetLength(long value)
+        {
+            ThrowIfAlreadyTimedOut();
+            
+            inner.SetLength(value);
+        }
 
-        public override void Write(byte[] buffer, int offset, int count) => inner.Write(buffer, offset, count);
+        public override void Write(byte[] buffer, int offset, int count)
+        {
+            TryCloseOnTimeout(() => inner.Write(buffer, offset, count));
+        }
 
-        public override void WriteByte(byte value) => inner.WriteByte(value);
+        public override void WriteByte(byte value)
+        {
+            TryCloseOnTimeout(() => inner.WriteByte(value));
+        }
 
 #if !NETFRAMEWORK
-        public override void CopyTo(Stream destination, int bufferSize) => inner.CopyTo(destination, bufferSize);
-        
-        public override int Read(Span<byte> buffer) => inner.Read(buffer);
+        public override void CopyTo(Stream destination, int bufferSize)
+        {
+            ThrowIfAlreadyTimedOut();
+            
+            inner.CopyTo(destination, bufferSize);
```

Comment:
Not sure this is correct.

We probably do need to be doing a `TryCloseOnTimeout` of course we wont know which stream had the timeout. 

I think to get around that the solution would be to do, `base.CopyTo` since that would then make calls to `read` on `this` which would then do the `TryCloseOnTimeout` work.